services/github/schemas/v1/taskcluster-render-yaml-input.yml

```diff
@@ -0,0 +1,76 @@
+$schema: "/schemas/common/metaschema.json#"
+title: "Render .taskcluster.yml input"
+description: |
+  Render .taskcluster.yml for given event.
+type:         object
+properties:
+  body:
+    title: Taskcluster.yml contents
+    description: The contents of the .taskcluster.yml
+    type: string
+  organization:
+    type: string
+    description: The organization that owns the repository.
+    pattern: "^[a-zA-Z0-9_-]+$"
+    default: TaskclusterRobot
+  repository:
+    type: string
+    description: The repository name.
+    pattern: "^[a-zA-Z0-9_-]+$"
+    default: hooks-testing
+  branch:
+    type: string
+    description: The branch name.
+    default: main
+  fakeEventType:
+    type: string
+    description: Name of the emulated event type
+    enum:
+      - github-push
+      - github-tag-push
+      - github-pull-request
+      - github-pull-request-untrusted
+      - github-release
+  fakePullRequestAction:
+    type: string
+    description: Name of the pull request action
+    default: opened
+    enum:
+      - opened
+      - synchronize
+      - reopened
+      - assigned
+      - auto_merge_disabled
+      - auto_merge_enabled
+      - closed
+      - converted_to_draft
+      - dequeued
+      - edited
+      - enqueued
+      - labeled
+      - ready_for_review
+      - review_requested
+      - review_request_removed
+      - unassigned
+      - unlabeled
+  fakeReleaseAction:
+    type: string
+    description: Name of the release action
+    default: published
```

Comment:
curious about the choice of defaults overall. I'm not sure it doesn't complicate things without asking users to make an explicit choice.

Reply:
good question! My assumption was that it should be easy to test generic events like

```
http POST http://taskcluster/api/github/v1/taskcluster-yml body=@.taskcluster.yml fakeEventType=github-push
http POST http://taskcluster/api/github/v1/taskcluster-yml body=@.taskcluster.yml fakeEventType=github-pull-request
```
and if you need to be more specific you could 
```
http POST http://taskcluster/api/github/v1/taskcluster-yml body=@.taskcluster.yml fakeEventType=github-pull-request fakePullRequestAction=synchronize
```

But I see your point. We can also require all of it and don't assume any defaults at all 🤷